datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -624,11 +645,13 @@ public AbstractKafkaBasedConnectorTask getConnectorTask() {
   public Map<String, Map<Integer, Long>> getConsumptionLagForTask(DatastreamTask task) throws IllegalArgumentException {
     Validate.notNull(task);
 
-    ConnectorTaskEntry taskEntry = _runningTasks.getOrDefault(task, null);
-    if (taskEntry == null) {
-      throw new IllegalArgumentException("Task not found.");
-    }
+    synchronized (_runningTasks) {
```

Comment:
Clarified with @surajkn. This change is not related to the bug that's being fixed. This is just extra synchronization code.